HIP/hip-1068.md

```diff
@@ -0,0 +1,528 @@
+```protobuf
+    message QueryHeader {
+        ...
+
+        /**
+         * The account a spender is noting has provided prior willingness through an HBAR allowance to sponsor the
+         * query transaction fees
+         */
+        AccountID sponsor_account_id = 3;
+    }
+```
+
+> Note: A user must specify the sponsor_account_id for the network to consider the validity of the sponsored transaction. 
+If not specified, the network will treat this as a normal transaction in which the submitter is due the transaction
+fees.
+
+### Error Cases
+
+As with normal transaction it’s possible for a sponsored transaction to fail.
+Existing transaction network rules will be applicable. 
+
+A node that encounters an invalid sponsoring account value will respond with a new response code
+
+```protobuf
+    enum ResponseCodeEnum {
+        ...
+
+        INVALID_SPONSOR_ACCOUNT = 352;
+    }
+```
+
+Transaction sponsorship will be complete in nature, a sponsor must be able to cover the full fee.
+In the case a sponsor lacks insufficient balance to cover an account that transaction will fail with a new response
+code.
+
+```protobuf
+    enum ResponseCodeEnum {
+        ...
+        
+        INSUFFICIENT_SPONSOR_ACCOUNT_BALANCE = 353;
+    }
+```
+
+### Transaction Id
+
+Currently the transaction id of any submitted transaction takes the form of `<accountId>@<validStartTime>` format.
+Usually the `accountId` signifies the user who both submitted and paid for the transaction. With the changes from this
+HIP the `accountId` value in the transaction id of a sponsored transaction will refer only to the submitter and no
+longer the payer. The accountId of the payer will be located in the transaction body. 
+
+### System Contract Functions
+
+Similar to [HIP 906](https://hips.hedera.com/hip/hip-906), the system contract interface would need to be updated to
+support the management of allowances with scope by a smart contract.
+
+The following functions are introduced in this HIP
+```solidity
+    interface IHRC1068 {
+        /// extensions of HIP 514
+        function approve(address token, address spender, uint256 amount, AllowanceScope allowanceScope) external
+            returns (int64 responseCode);
+        function approveNFT(address token, address approved, uint256 serialNumber, AllowanceScope allowanceScope)
+            external returns (int64 responseCode);
+        function setApprovalForAll(address token, address operator, bool approved, AllowanceScope allowanceScope)
+            external returns (int64 responseCode);
+
+        /// extensions of HIP 906
+        function hbarAllowance(address spender, AllowanceScope allowanceScope) external returns (responseCode,
+            int256 amount);
+        function hbarApprove(address spender, int256 amount, AllowanceScope allowanceScope) external
+            returns (responseCode);
+    }
+```
+
+`IHRC1068` token methods will be callable via the HTS system contract at `0x167`.
```

Comment:
While I know HTS I do not know what HAS or EOA stand for. Can we add a link or the full name at first usage?

Reply:
HAS -> Hedera Account Service
EAO -> Externally owned account

I agree with your comment for adding more info about the full names.

Reply:
Added